contracts/modules/ComponentController.sol

```diff
@@ -196,13 +194,20 @@ contract ComponentController is
     }
 
     function getComponentType(uint256 id) public view returns (IComponent.ComponentType componentType) {
-        return _componentType[id];
+        IComponent component = _componentById[id];
+        componentType = component.getType();
     }
 
     function getComponentState(uint256 id) public view returns (IComponent.ComponentState componentState) {
         return _componentState[id];
     }
 
+    function getRequiredRole(IComponent.ComponentType componentType) external returns (bytes32) {
+        if (componentType == IComponent.ComponentType.Product) { return _access.productOwnerRole(); }
+        if (componentType == IComponent.ComponentType.Oracle) { return _access.oracleProviderRole(); }
+        if (componentType == IComponent.ComponentType.Riskpool) { return _access.riskpoolKeeperRole(); }

```

Comment:
change to if/if else/else and throw exception if other component type (you never know).